Sync `css/css-variables` from WPT upstream

Ahmad-S792 opened the pull request:
#### 09d8d82ba20934b7bd6756fc044d24177dbcbf76
<pre>
Sync `css/css-variables` from WPT upstream

<a href="https://bugs.webkit.org/show_bug.cgi?id=294576">https://bugs.webkit.org/show_bug.cgi?id=294576</a>
<a href="https://rdar.apple.com/153568924">rdar://153568924</a>

Reviewed by Anne van Kesteren.

Upstream commit: <a href="https://github.com/web-platform-tests/wpt/commit/2f8491c9b68ba98953266654763398485905f4de">https://github.com/web-platform-tests/wpt/commit/2f8491c9b68ba98953266654763398485905f4de</a>

* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/revert-in-fallback-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/revert-in-fallback.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/variable-cycles-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/variable-cycles.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/variable-definition-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/variable-definition.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/variable-generated-content-dynamic-001.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/variable-substitution-variable-declaration-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/variable-substitution-variable-declaration.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/reference/variable-reference-without-whitespace-ref.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/reference/vars-background-shorthand-001-ref.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/reference/vars-font-shorthand-001-ref.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/reference/w3c-import.log
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/url-syntax-crash.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/var-parsing-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/var-parsing.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/variable-css-wide-keywords-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/variable-css-wide-keywords.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/variable-invalid-recovery-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/variable-invalid-recovery.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/variable-recalc-with-initial-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/variable-recalc-with-initial.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/variables-animation-math-functions-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/variables-animation-math-functions.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/WEB_FEATURES.yml:
* LayoutTests/TestExpectations: Add &apos;Crashing&apos; test on Debug as &apos;Skip&apos; with filed bug
* LayoutTests/platform/glib/TestExpectations: Ditto

Canonical link: <a href="https://commits.webkit.org/296314@main">https://commits.webkit.org/296314@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b75ff84385fe33052bc52b5b066b2fcae4045044

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/108147 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/27810 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/18230 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/113358 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/58633 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/28508 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/36360 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/82112 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/111095 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/22585 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/97422 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/62543 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/21999 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/15558 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/58098 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/91949 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/15614 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/116484 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/35213 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/25929 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/91139 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/35587 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/93699 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/90934 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23173 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/35822 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/13585 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/31011 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/35112 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/34846 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/38204 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/36513 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/22591f73c6bbf9df9fd505529c220a054bba3f41. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/46819

Ahmad-S792 committed:
Sync `css/css-variables` from WPT upstream

https://bugs.webkit.org/show_bug.cgi?id=294576
rdar://153568924

Reviewed by Anne van Kesteren.

Upstream commit: https://github.com/web-platform-tests/wpt/commit/2f8491c9b68ba98953266654763398485905f4de

* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/revert-in-fallback-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/revert-in-fallback.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/variable-cycles-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/variable-cycles.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/variable-definition-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/variable-definition.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/variable-generated-content-dynamic-001.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/variable-substitution-variable-declaration-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/variable-substitution-variable-declaration.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/reference/variable-reference-without-whitespace-ref.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/reference/vars-background-shorthand-001-ref.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/reference/vars-font-shorthand-001-ref.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/reference/w3c-import.log
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/url-syntax-crash.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/var-parsing-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/var-parsing.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/variable-css-wide-keywords-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/variable-css-wide-keywords.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/variable-invalid-recovery-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/variable-invalid-recovery.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/variable-recalc-with-initial-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/variable-recalc-with-initial.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/variables-animation-math-functions-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/variables-animation-math-functions.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/WEB_FEATURES.yml:
* LayoutTests/TestExpectations: Add 'Crashing' test on Debug as 'Skip' with filed bug
* LayoutTests/platform/glib/TestExpectations: Ditto

Canonical link: https://commits.webkit.org/296314@main
